build/builder_test.go

```diff
@@ -455,67 +456,120 @@ func TestBuilder_DeltaShardsBuildsShouldErrorOnBranchSet(t *testing.T) {
 	}
 }
 
-func TestBuilder_DeltaShardsUpdateVersionsInOlderShards(t *testing.T) {
-	indexDir := t.TempDir()
+func TestBuilder_DeltaShardsMetadataInOlderShards(t *testing.T) {
+	olderTime := time.Unix(0, 0)
+	newerTime := time.Unix(10000, 0)
 
-	repositoryV1 := zoekt.Repository{
-		Name: "repo",
-		ID:   1,
-		Branches: []zoekt.RepositoryBranch{
-			{Name: "main", Version: "v1"},
-			{Name: "release", Version: "v1"},
+	for _, test := range []struct {
+		name               string
+		originalRepository zoekt.Repository
+		updatedRepository  zoekt.Repository
+	}{
+		{
+			name: "update commit information",
+			originalRepository: zoekt.Repository{
+				Name: "repo",
+				ID:   1,
+				Branches: []zoekt.RepositoryBranch{
+					{Name: "main", Version: "v1"},
+					{Name: "release", Version: "v1"},
+				},
+			},
+			updatedRepository: zoekt.Repository{
+				Name: "repo",
+				ID:   1,
+				Branches: []zoekt.RepositoryBranch{
+					{Name: "main", Version: "v2"},
+					{Name: "release", Version: "v2"},
+				},
+			},
 		},
-	}
-
-	createTestShard(t, indexDir, repositoryV1, 2, func(o *Options) {
-		o.DisableCTags = true
-	})
-
-	repositoryV2 := zoekt.Repository{
-		Name: "repo",
-		ID:   1,
-		Branches: []zoekt.RepositoryBranch{
-			{Name: "main", Version: "v2"},
-			{Name: "release", Version: "v2"},
+		{
+			name: "update latest commit date (older -> newer)",
+			originalRepository: zoekt.Repository{
+				Name: "repo",
+				ID:   1,
+				Branches: []zoekt.RepositoryBranch{
+					{Name: "main", Version: "v1"},
+				},
+				LatestCommitDate: olderTime,
+			},
+			updatedRepository: zoekt.Repository{
+				Name: "repo",
+				ID:   1,
+				Branches: []zoekt.RepositoryBranch{
+					{Name: "main", Version: "v2"},
+				},
+				LatestCommitDate: newerTime,
+			},
 		},
-	}
+		{
+			name: "update latest commit date (even if latest commit date is older)",
```

Comment:
How can a delta shard have an older date? I assume in case of force pushes we anyway reindex from scratch?

Reply:
if we are getting the commit date info from git, then it is from an untrusted source and could be any possible date.